lld/wasm/InputChunks.h

```diff
@@ -259,10 +259,13 @@ class InputFunction : public InputChunk {
         file->codeSection->Content.slice(inputSectionOffset, function->Size);
     debugName = function->DebugName;
     comdat = function->Comdat;
+    assert(s.Kind != WasmSignature::Placeholder);
```

Comment:
Should we give a more useful error message here? 

Perhaps we simply not set `isRelocatableObject()` for any object file that contains any of these.  Then the linker would simply error on them before getting this far?

Reply:
My guess with the assert was that we should never get here, because we would never expect to see a file with a symbol table that also had GC. But yeah, also enforcing it on entry to the linker makes sense. I wonder if a more specific or meaningful error message would make sense (I guess in this case if you tried to link something that just happened to have some unrelated GC types, the link error would be the same as if there were no linking section).

Reply:
Yes, perhaps another flag in the object file, ("hasGCTypes").. or perhaps the linker to iterate through the types section to make sure it understands them all and error out if it doesn't?

Reply:
I added a more explicit check in `createObjectFile`. I don't have test because I haven't yet successfully created a file with both GC types and a linking section that wasn't also rejected by the object file parser. But a little redundancy in the checks seems fine.